### ISSUE TITLE
Bug: `mono-src` is required

### ISSUE THREAD

**Issue by AmitMY** [mozilla/firefox-translations-training#108]
I don't think `mono-src` datasets should be required, nor training a backward model. 
While this repo is designed mostly for high-resource languages, there are also extreme low-resource use cases I see and want to do.

**Comment by eu9ene** [comment]
`mono-src` is always required because you need to augment the training corpus with a lot of clean monolingual data to train the student with minimal quality loss. Monolingual data is easily accessible for all languages we're interested in including low resource (if it's not, you can't train anything useful).

A backward model is required for the cross-entropy filtering step and is crucial for training low-resource languages. Teacher training data augmentation might be skipped for extremely high-resource languages like French but is still recommended. The model learns how to translate low-quality noisy text to clean text with proper grammar.

The repo is designed for both high and low-resource languages. What language are you working on? What issues do you have with your use case?

Closing this one as a non-issue.

**Comment by AmitMY** [comment]
So just for context, I work with written forms of signed languages. 
You can write signed languages in a phonetic writing system like SignWriting or HamNoSys (see [PDF](https://sign-language-processing.github.io/assets/representation/continuous.pdf))

This "IPA" like form rarely exists by itself, without parallel data, so signed languages have a different use case.

**Perhaps then what I should do is:**
1. Write a sign language transcription model - given a video, write it down as text (would have loved to use bergamot for that, but I'm afraid that would be too complicated - `vectors-to-text` rather than `text-to-text`)
2. Include all this text as monolingual data. But since it is more often than not parallel to spoken language data, this method will generate a lot more parallel data than monolingual.

In any case, I understand your points, just thought I would lay my use case so it is more clear.

**Comment by eu9ene** [comment]
Thanks for the context! This is a very interesting use case and I think you can work around this and skip the backward training model and augmentation. However, the end to end pipeline is still designed to produce a compressed model and for this, we always need to get more source monolingual data. You can try to remove this step from snakefile if you don't have it.
